src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts

```diff
@@ -46,8 +46,15 @@ export class PlagiarismCaseInstructorDetailViewComponent implements OnInit, OnDe
                 this.createEmptyPost();
             },
         });
-        this.postsSubscription = this.metisService.posts.pipe().subscribe((posts: Post[]) => {
-            this.posts = posts;
+        this.postsSubscription = this.metisService.posts.subscribe((posts: Post[]) => {
+            const filteredPosts = posts?.filter((post) => post.plagiarismCase?.id === this.plagiarismCaseId);
```

Comment:
We don't need that here. If ``posts`` were a falsy value, it would throw an error in line 53 as you check the length of undefined references there. If you want to check it, just add a
```js
if (!posts) (
    return;
}
```
---
```suggestion
            const filteredPosts = posts.filter((post) => post.plagiarismCase?.id === this.plagiarismCaseId);
```

Reply:
Makes sense, thanks 👍